### ISSUE TITLE
Not able to Download contents

### ISSUE THREAD

**Issue by jdsingh253** [r0oth3x49/acloud-dl#90]
#######################Getting Error#######################
[*] : Course 'AWS Certified Solutions Architect - Associate (SAA-C02)'.
[*] : Downloaded course information .. (done)
[+] : Chapter(s) (23).
[*] : Lecture(s) (201).

[*] : Downloading chapter : (1 of 23)
[+] : Chapter (01 Introduction)
[*] : Found (4) lectures ...

[*] : Lecture(s) : (1 of 4)
[*] : Downloading (001 Course Introduction)
[*] : 84.67MB/84.67MB 100.00% |##############################| 4.12MB/s
[+] : Downloaded  (001 Course Introduction)

[*] : Downloading asset(s)
[*] : Downloading (001 Study Guide (PDF) Course Slides.pdf)
[*] : 544.93MB/544.93MB 100.00% |##############################| 4.91MB/s
[+] : Downloaded  (001 Study Guide (PDF) Course Slides.pdf)

[*] : Lecture(s) : (2 of 4)
[*] : Downloading (002 About the Training Architect)
[*] : 65.16MB/65.16MB 100.00% |##############################| 4.81MB/s
[+] : Downloaded  (002 About the Training Architect)

[*] : Lecture(s) : (3 of 4)
[*] : Downloading (003 Exam Guide)
[*] : 78.05MB/78.05MB 100.00% |##############################| 5.36MB/s
[+] : Downloaded  (003 Exam Guide)

[*] : Downloading asset(s)
[*] : Downloading (003 Exam Guide.pdf)
[*] : 152.40KB/152.40KB 100.00% |##############################| 761.13kB/s
[+] : Downloaded  (003 Exam Guide.pdf)

[*] : Lecture(s) : (4 of 4)
[*] : Downloading (004 Course Updates)
[*] : 17.91MB/17.91MB 100.00% |##############################| 5.45MB/s
[+] : Downloaded  (004 Course Updates)

[*] : Downloading chapter : (2 of 23)
[+] : Chapter (02 Supporting Courses)
[*] : Found (7) lectures ...

[*] : Lecture(s) : (1 of 7)
[*] : Downloading (001 Certification Path and Supporting Courses)
[*] : 22.78MB/22.78MB 100.00% |##############################| 5.14MB/s
[+] : Downloaded  (001 Certification Path and Supporting Courses)

[*] : Downloading asset(s)
[*] : Downloading (001 AWS Certified Solutions Architect Associate Certification Path.pdf)
[*] : 4.43MB/4.43MB 100.00% |##############################| 2.28MB/s
[+] : Downloaded  (001 AWS Certified Solutions Architect Associate Certification Path.pdf)

[*] : Lecture(s) : (2 of 7)
[*] : Downloading (002 Designing Resilient Architectures for Associate AWS Solutions Architects)
[*] : 26.12MB/26.12MB 100.00% |##############################| 5.19MB/s
[+] : Downloaded  (002 Designing Resilient Architectures for Associate AWS Solutions Architects)

[*] : Downloading asset(s)
[*] : Downloading (002 AWS Certified Solutions Architect Associate Certification Path.pdf)
[*] : 4.43MB/4.43MB 100.00% |##############################| 2.34MB/s
[+] : Downloaded  (002 AWS Certified Solutions Architect Associate Certification Path.pdf)

[*] : Lecture(s) : (3 of 7)
[*] : Downloading (003 Networking and Compute for Associate AWS Solutions Architects)
[*] : 23.23MB/23.23MB 100.00% |##############################| 5.40MB/s
[+] : Downloaded  (003 Networking and Compute for Associate AWS Solutions Architects)

[*] : Downloading asset(s)
[*] : Downloading (003 AWS Certified Solutions Architect Associate Certification Path.pdf)
[*] : 4.43MB/4.43MB 100.00% |##############################| 2.10MB/s
[+] : Downloaded  (003 AWS Certified Solutions Architect Associate Certification Path.pdf)

[*] : Lecture(s) : (4 of 7)
[*] : Downloading (004 Storage Databases and Migration for Associate AWS Solutions Architects)
[*] : 23.97MB/23.97MB 100.00% |##############################| 4.99MB/s
[+] : Downloaded  (004 Storage Databases and Migration for Associate AWS Solutions Architects)

[*] : Downloading asset(s)
[*] : Downloading (004 AWS Certified Solutions Architect Associate Certification Path.pdf)
[*] : 4.43MB/4.43MB 100.00% |##############################| 3.45MB/s
[+] : Downloaded  (004 AWS Certified Solutions Architect Associate Certification Path.pdf)

[*] : Lecture(s) : (5 of 7)
[*] : Downloading (005 High Availability and Scalability for Associate AWS Solutions Architects)
[*] : 22.60MB/22.60MB 100.00% |##############################| 5.42MB/s
[+] : Downloaded  (005 High Availability and Scalability for Associate AWS Solutions Architects)

[*] : Downloading asset(s)
[*] : Downloading (005 AWS Certified Solutions Architect Associate Certification Path.pdf)
[*] : 4.43MB/4.43MB 100.00% |##############################| 2.20MB/s
[+] : Downloaded  (005 AWS Certified Solutions Architect Associate Certification Path.pdf)

[*] : Lecture(s) : (6 of 7)
[*] : Downloading (006 Application Services for Associate AWS Solutions Architects)
[*] : 23.82MB/23.82MB 100.00% |##############################| 4.65MB/s
[+] : Downloaded  (006 Application Services for Associate AWS Solutions Architects)

[*] : Downloading asset(s)
[*] : Downloading (006 AWS Certified Solutions Architect Associate Certification Path.pdf)
[*] : 4.43MB/4.43MB 100.00% |##############################| 2.25MB/s
[+] : Downloaded  (006 AWS Certified Solutions Architect Associate Certification Path.pdf)

[*] : Lecture(s) : (7 of 7)
[*] : Downloading (007 Logging and Security for Associate AWS Solutions Architects)
[*] : 23.57MB/23.57MB 100.00% |##############################| 5.62MB/s
[+] : Downloaded  (007 Logging and Security for Associate AWS Solutions Architects)

[*] : Downloading asset(s)
[*] : Downloading (007 AWS Certified Solutions Architect Associate Certification Path.pdf)
[*] : 4.43MB/4.43MB 100.00% |##############################| 2.29MB/s
[+] : Downloaded  (007 AWS Certified Solutions Architect Associate Certification Path.pdf)

[*] : Downloading chapter : (3 of 23)
[+] : Chapter (03 AWS Fundamentals)
[*] : Found (6) lectures ...

[*] : Lecture(s) : (1 of 6)
[*] : Downloading (001 The Building Blocks of AWS Availability Zones and Regions)
[*] : 187.75MB/187.75MB 100.00% |##############################| 4.65MB/s
[+] : Downloaded  (001 The Building Blocks of AWS Availability Zones and Regions)

[*] : Lecture(s) : (2 of 6)
[*] : Downloading (002 Who Owns What in the Cloud)
[*] : 62.23MB/62.23MB 100.00% |##############################| 4.44MB/s
[+] : Downloaded  (002 Who Owns What in the Cloud)

[*] : Lecture(s) : (3 of 6)
[*] : Downloading (003 Compute Storage Databases and Networking)
[*] : 81.00MB/81.00MB 100.00% |##############################| 4.29MB/s
[+] : Downloaded  (003 Compute Storage Databases and Networking)

[*] : Lecture(s) : (4 of 6)
[*] : Downloading (004 What Is the Well-Architected Framework)
[*] : 54.25MB/54.25MB 100.00% |##############################| 3.44MB/s
[+] : Downloaded  (004 What Is the Well-Architected Framework)

[*] : Lecture(s) : (5 of 6)
[*] : Downloading (005 AWS Fundamentals Exam Tips)
[*] : 34.35MB/34.35MB 100.00% |##############################| 4.17MB/s
[+] : Downloaded  (005 AWS Fundamentals Exam Tips)

[*] : Downloading chapter : (4 of 23)
[+] : Chapter (04 Identity and Access Management (IAM))
[*] : Found (7) lectures ...

[*] : Lecture(s) : (1 of 7)
[*] : Downloading (001 Securing the Root Account)
[*] : 92.20MB/92.20MB 100.00% |##############################| 2.67MB/s
[+] : Downloaded  (001 Securing the Root Account)

[*] : Lecture(s) : (2 of 7)
[*] : Downloading (002 Controlling Users Actions with IAM Policy Documents)
[*] : 101.50MB/101.50MB 100.00% |##############################| 2.74MB/s
[+] : Downloaded  (002 Controlling Users Actions with IAM Policy Documents)

[*] : Lecture(s) : (3 of 7)
[*] : Downloading (003 Permanent IAM Credentials)
[*] : 280.73MB/280.73MB 100.00% |##############################| 4.33MB/s
[+] : Downloaded  (003 Permanent IAM Credentials)

[*] : Lecture(s) : (4 of 7)
[*] : Downloading (004 IAM Exam Tips)
[*] : 52.52MB/52.52MB 100.00% |##############################| 5.75MB/s
[+] : Downloaded  (004 IAM Exam Tips)

[*] : Downloading chapter : (5 of 23)
[+] : Chapter (05 Simple Storage Service (S3))
[*] : Found (15) lectures ...

[*] : Lecture(s) : (1 of 15)
[*] : Downloading (001 S3 Overview)
[*] : 223.69MB/223.69MB 100.00% |##############################| 4.55MB/s
[+] : Downloaded  (001 S3 Overview)

[*] : Lecture(s) : (2 of 15)
[*] : Downloading (002 Securing Your Bucket with S3 Block Public Access)
[*] : 81.07MB/81.07MB 100.00% |##############################| 4.40MB/s
[+] : Downloaded  (002 Securing Your Bucket with S3 Block Public Access)

[*] : Lecture(s) : (3 of 15)
[*] : Downloading (003 Hosting a Static Website Using S3)
[*] : 74.74MB/74.74MB 100.00% |##############################| 5.27MB/s
[+] : Downloaded  (003 Hosting a Static Website Using S3)

[*] : Downloading asset(s)
[*] : Downloading (003 Html  JSON Files.zip)
[*] : 2.01KB/2.01KB 100.00% |##############################| 334.31kB/s
[+] : Downloaded  (003 Html  JSON Files.zip)

[*] : Lecture(s) : (4 of 15)
[*] : Downloading (004 Versioning Objects in S3)
[*] : 67.87MB/67.87MB 100.00% |##############################| 5.60MB/s
[+] : Downloaded  (004 Versioning Objects in S3)

[*] : Lecture(s) : (5 of 15)
[*] : Downloading (005 S3 Storage Classes)
[*] : 115.84MB/115.84MB 100.00% |##############################| 5.29MB/s
[+] : Downloaded  (005 S3 Storage Classes)

[*] : Downloading asset(s)
[*] : Downloading (005 Retrieving S3 Glacier Archives.html)
[+] : Downloaded  (005 Retrieving S3 Glacier Archives.html)

[*] : Lecture(s) : (6 of 15)
[*] : Downloading (006 Lifecycle Management with S3)
[*] : 48.59MB/48.59MB 100.00% |##############################| 6.01MB/s
[+] : Downloaded  (006 Lifecycle Management with S3)

[*] : Lecture(s) : (7 of 15)
[*] : Downloading (007 S3 Object Lock and Glacier Vault Lock)
[*] : 72.86MB/72.86MB 100.00% |##############################| 5.38MB/s
[+] : Downloaded  (007 S3 Object Lock and Glacier Vault Lock)

[*] : Lecture(s) : (8 of 15)
[*] : Downloading (008 Encrypting S3 Objects)
[*] : 103.62MB/103.62MB 100.00% |##############################| 3.74MB/s
[+] : Downloaded  (008 Encrypting S3 Objects)

[*] : Lecture(s) : (9 of 15)
[*] : Downloading (009 Optimizing S3 Performance)
[*] : 74.95MB/74.95MB 100.00% |##############################| 5.44MB/s
[+] : Downloaded  (009 Optimizing S3 Performance)
Traceback (most recent call last):
  File "C:\Users\jdsingh\Documents\acloud-dl-master\acloud-dl.py", line 408, in <module>
    main()
  File "C:\Users\jdsingh\Documents\acloud-dl-master\acloud-dl.py", line 396, in main
    cloud_guru.course_download(path=options.output, quality=options.quality,
  File "C:\Users\jdsingh\Documents\acloud-dl-master\acloud-dl.py", line 295, in course_download
    self.download_lectures_only(lecture_best=lecture_best, lecture_title=lecture_title,
  File "C:\Users\jdsingh\Documents\acloud-dl-master\acloud-dl.py", line 215, in download_lectures_only
    self.download_assets(lecture_assets=lecture_assets, filepath=filepath)
  File "C:\Users\jdsingh\Documents\acloud-dl-master\acloud-dl.py", line 156, in download_assets
    assets.download(filepath=filepath, quiet=True, callback=self.show_progress)
  File "C:\Users\jdsingh\Documents\acloud-dl-master\acloud\_shared.py", line 857, in download
    return self._write_external_links(filepath)
  File "C:\Users\jdsingh\Documents\acloud-dl-master\acloud\_shared.py", line 773, in _write_external_links
    with open("{}".format(filename), "a", encoding="utf-8") as f:
OSError: [Errno 22] Invalid argument: '\\\\?\\\\\\?\\C:\\Users\\jdsingh\\Documents\\acloud-dl-master\\AWS Certified Solutions Architect - Associate (SAA-C02)\\05 Simple Storage Service (S3)\\009 For Amazon S3 request rates whats the difference between prefixes and nested folders How many prefixes can I have in an S3 bucket.txt'
PS C:\Users\jdsingh\Documents\acloud-dl-master>

**Comment by Optimus50** [comment]
seems this project is also dead

**Comment by r0oth3x49** [comment]
@Optimus50 this is not dead i am alive 😉 . i might need an account to reproduce and fix the issue. BTW sorry for late response.

**Comment by onoakino** [comment]
Whey you say you need an account, you mean a paid subscription account and not a free subscription account?

**Comment by skmd0** [comment]
@r0oth3x49 the issue is that the filename path is too long on Windows and the magic path prefix doesn't work correctly. Maybe the easiest solution is just to trim the filepath to less than 255 characters? I don't know python, so can't make a proper fix myself.

**Comment by r0oth3x49** [comment]
In latest windows you can handle this at the time of installation @skmd0 but I will check the code. Triming things is not really a good solution though

**Comment by r0oth3x49** [comment]
@onoakino yes and that is if you are okay to share otherwise I can't reproduce and fix the issue.

**Comment by skmd0** [comment]
I agree with you that triming is a bad solution. I did it locally just so I could download videos for myself, since this bug was preventing me to do so. I checked the downloaded files and I think only .txt files were affected because they have really long names.

Basically what I did was changing the `filepath = "\\\\?\\{}".format(filepath)` to `filepath = filepath[:250]` everywhere magic path is used (in `_shared.py` file). Quick and dirty solution, but that's as far as my python knowledge goes. I am sharing it here in case anyone wants a quick fix for themselves until the proper solution is done in master.
